src/kernel/boot.c

```diff
@@ -665,23 +665,24 @@ BOOT_CODE static bool_t provide_untyped_cap(
 
     /* All cap ptrs must be aligned to object size */
     if (!IS_ALIGNED(pptr, size_bits)) {
-        printf("Kernel init: Unaligned untyped pptr %p (alignment %"SEL4_PRIu_word")\n", (void *)pptr, size_bits);
+        printf("Kernel init: Unaligned untyped pptr %"SEL4_PRIu_word" (alignment %"SEL4_PRIu_word")\n", (word_t)pptr,
+               size_bits);
         return false;
     }
 
     /* All cap ptrs apart from device untypeds must be in the kernel window. */
     if (!device_memory && !pptr_in_kernel_window(pptr)) {
-        printf("Kernel init: Non-device untyped pptr %p outside kernel window\n",
-               (void *)pptr);
+        printf("Kernel init: Non-device untyped pptr %"SEL4_PRIu_word"  outside kernel window\n",
```

Comment:
I don't see this cast doing any harm - also %p will print 0x<addr> whereas you are now printing it in decimal?

Reply:
Not sure what type pptr_t is but if it's triggering a warning `(void *)(uintptr_t)pptr` is IMO cleaner.